CMakeLists.txt

```diff
@@ -392,6 +392,13 @@ ELSE(CMAKE_TOOLCHAIN_FILE)
   # Check that a C++ 14 standard library is configured
   include(CMake/TestCxx14Library.cmake)
 
+  #------------------------------
+  #  check Intel compiler version
+  #------------------------------
+  if ((${COMPILER} MATCHES "Intel") AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 18.0))
```

Comment:
Can we put this into Intel configuration file CMake/IntelCompilers.cmake?

Reply:
Is the chain rule of Intel using libstdc++ from gcc causing the so high version?
Do we also need to raise gcc and clang version minimum?

Reply:
No, there are specific bugs in Intel 17.

Reply:
You're right, we should update the minimum versions on clang and gcc for C++ 14 support.

Reply:
Shouldn't CMAKE_CXX_STANDARD take care of that for us? The whole point is to not have to fiddle with individual compilers. Sometimes it can't be avoided (e.g. like avoiding version-specific bugs in Intel), but it seems like it shouldn't be necessary just to guarantee standard support.

Reply:
Agreed.

The only reason to name specific compiler versions should be if we know they are problematically buggy.